src/main/java/frc/robot/poseestimator/WPILibPoseEstimator/WPILibPoseEstimator.java

```diff
@@ -0,0 +1,115 @@
+package frc.robot.poseestimator.WPILibPoseEstimator;
+
+import edu.wpi.first.math.VecBuilder;
+import edu.wpi.first.math.estimator.PoseEstimator;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.interpolation.TimeInterpolatableBuffer;
+import edu.wpi.first.math.kinematics.Odometry;
+import edu.wpi.first.math.kinematics.SwerveDriveKinematics;
+import edu.wpi.first.math.kinematics.SwerveDriveOdometry;
+import edu.wpi.first.math.kinematics.SwerveModulePosition;
+import frc.robot.poseestimator.IPoseEstimator;
+import frc.robot.poseestimator.observations.IRobotPoseVisionObservation;
+import frc.robot.poseestimator.observations.OdometryObservation;
+import frc.robot.subsystems.GBSubsystem;
+import frc.robot.vision.rawdata.RawAprilTagVisionData;
+
+import java.util.List;
+import java.util.Optional;
+
+public class WPILibPoseEstimator extends GBSubsystem implements IPoseEstimator {
+
+	private final PoseEstimator<SwerveModulePosition[]> poseEstimator;
+	private final Odometry<SwerveModulePosition[]> odometryEstimator;
+
+	private final TimeInterpolatableBuffer<Pose2d> timeInterpolatableBuffer;
+
+	public WPILibPoseEstimator(
+		String logPath,
+		SwerveDriveKinematics kinematics,
+		SwerveDriveOdometry odometry,
+		SwerveModulePosition[] modulePositions
+	) {
+		super(logPath);
+
+
+		this.poseEstimator = new PoseEstimator<>(
+			kinematics,
+			odometry,
+			WPILibPoseEstimatorConstants.DEFAULT_ODOMETRY_STANDARD_DEVIATIONS.getWPILibStandardDeviations(),
+			WPILibPoseEstimatorConstants.DEFAULT_VISION_STANDARD_DEVIATIONS.getWPILibStandardDeviations()
+		);
+		this.odometryEstimator = new Odometry<>(
+			kinematics,
+			WPILibPoseEstimatorConstants.STARTING_ODOMETRY_ANGLE,
+			modulePositions,
+			WPILibPoseEstimatorConstants.STARTING_ODOMETRY_POSE
+		);
+
+		this.timeInterpolatableBuffer = TimeInterpolatableBuffer.createBuffer(WPILibPoseEstimatorConstants.POSE_BUFFER_SIZE_SECONDS);
+		;
+	}
+
+
+	@Override
+	public void resetPose(Pose2d newPose) {
+		poseEstimator.resetPose(newPose);
+	}
+
+	@Override
+	public Pose2d getEstimatedPose() {
+		return poseEstimator.getEstimatedPosition();
+	}
+
+	@Override
+	public Pose2d getEstimatedPoseAtTimestamp(double timestamp) {
+		return timeInterpolatableBuffer.getSample(timestamp).orElseGet(this::getEstimatedPose);
+	}
```

Comment:
there is a method for obtaining the estimated pose at a timestamp in the WPIlib PoseEstimator